keycloak_user_export/management/commands/migrate_users_to_keycloak.py

```diff
@@ -0,0 +1,224 @@
+                    "totp": False,
+                    "emailVerified": True,
+                    "email": user.email,
+                    "credentials": [],
+                    "disableableCredentialTypes": [],
+                    "requiredActions": [],
+                    "realmRoles": ["default-roles-master"],
+                    "notBefore": 0,
+                    "groups": [],
+                }
+                # If the user has a password defined, we will create a credential for them in Keycloak.
+                # This allows the user to
+                if user.password and user.has_usable_password():
+                    _, iterations, salt, hash = user.password.split("$", 3)
+                    base64_salt = base64.b64encode(salt.encode())
+                    user_keycloak_payload["credentials"].append(
+                        {
+                            "secretData": json.dumps(
+                                {"value": hash, "salt": base64_salt.decode()}
+                            ),
+                            "type": "password",
+                            "credentialData": json.dumps(
+                                {
+                                    "hashIterations": iterations,
+                                    "algorithm": "pbkdf2-sha256",
+                                }
+                            ),
+                        }
+                    )
+                unsynced_users_array.append(user_keycloak_payload)
+            headers = {
+                "Content-Type": "application/json",
+                "Authorization": f"Bearer {access_token}",
+            }
+            test_payload = json.dumps(
+                {
+                    "ifResourceExists": "SKIP",
+                    "realm": "master",
+                    "users": unsynced_users_array,
+                }
+            )
+            response = requests.request("POST", url, headers=headers, data=test_payload)
+            # If Keycloak responds with a 401, refresh the access_token and retry once.
+            if response.status_code == 401:
+                access_token = self.get_access_token(
+                    kwargs["client_id"],
+                    kwargs["email"],
+                    kwargs["password"],
+                    kwargs["client_secret"],
+                )
+                headers = {
+                    "Content-Type": "application/json",
+                    "Authorization": f"Bearer {access_token}",
+                }
+                response = requests.request(
+                    "POST", url, headers=headers, data=test_payload
+                )
+
+            # If the response from Keycloak is not 200, print out an error and move on to
+            # the next batch of users to export.
+            if response.status_code != 200:
+                print(
+                    f"Error calling Keycloak REST API, returned {response.status_code}"
+                )
```

Comment:
Should this output the body of the response? Status code alone might not be detailed enough